src/cholupdates/rank_1/_seeger.py

```diff
@@ -0,0 +1,262 @@
+"""Interface functions for the rank-1 up- and downdate algorithms specified in sections
+2 and 3 of [1].
+
+References
+----------
+.. [1] M. Seeger, "Low Rank Updates for the Cholesky Decomposition", 2008.
+"""
+
+from typing import Optional
+
+import numpy as np
+
+from ._arg_validation import _validate_update_args
+from ._seeger_impl_python import update as _update_impl_python
+
+_update_available_impls = ["python"]
+
+try:
+    from ._seeger_impl_cython import update as _update_impl_cython
+
+    _cython_available = True
+    _update_available_impls.append("cython")
+
+    _update_impl_default = _update_impl_cython
+except ImportError:
+    _cython_available = False
+
+    _update_impl_default = _update_impl_python
+
+
+def update_seeger(
+    L: np.ndarray,
+    v: np.ndarray,
+    check_diag: bool = True,
+    overwrite_L: bool = False,
+    overwrite_v: bool = False,
+    impl: Optional[str] = None,
+) -> np.ndarray:
+    r"""Update a Cholesky factorization after addition of a positive semidefinite
+    symmetric rank-1 matrix using the algorithm from section 2 of [1]_.
+
+    In other words, given :math:`A = L L^T \in \mathbb{R}^{N \times N}` and
+    :math:`v \in \mathbb{R}^N`, compute :math:`L'` such that
+
+    .. math::
+        A' := A + v v^T = L' L'^T.
+
+    This function computes the Cholesky factorization of :math:`A'` from :math:`L` in
+    :math:`O(N^2)` time, which is faster than the :math:`O(N^3)` time complexity of
+    naively applying a Cholesky algorithm to :math:`A'` directly.
+
+    Parameters
+    ----------
+    L :
+        Lower-triangular Cholesky factor of :math:`A` with shape :code:`(N, N)` and
+        dtype :class:`numpy.double`.
+        Must have a non-zero diagonal.
```

Comment:
Technically speaking the algorithm supports strictly negative diagonal elements. Do you think I should still restrict this to positive elements?

Reply:
Ah I see. There is no need to assume you are given a Cholesky factor initially. It could be a general lower triangular matrix root as long as you can do Givens rotations, right? Then I would not change the docs.

Reply:
Yup, that's the idea